docs/pages/database-access/guides/mysql-cloudsql.mdx

```diff
@@ -0,0 +1,292 @@
+  The default "Cloud SQL Admin" IAM role includes more permissions than the
+  database agent needs to generate one-time user passwords. To further restrict
+  the service account, you can create a role that includes only the following
+  permissions:
+  ```text
+  cloudsql.users.list
+  cloudsql.users.update
+  ```
+</Admonition>
+
+Once created, go to that service account's Keys tab and create a new key:
+
+![Service Account Keys](../../../img/database-access/guides/cloudsql/service-account-keys@2x.png)
+
+Make sure to choose JSON format:
+
+![Service Account New Key](../../../img/database-access/guides/cloudsql/service-account-new-key@2x.png)
+
+Save the file, your Teleport Database Service will need to use it as GCP
+application credentials file.
+
+## Gather Cloud SQL Instance Information
+
+To connect a Cloud SQL database to Teleport, you'll need to gather a few pieces
+of information about the instance.
+
+- GCP Project ID.
+
+You can normally see it in the organization view at the top of the GCP dashboard.
+
+- Cloud SQL instance ID.
+
+The instance ID is the name of your Cloud SQL instance shown at the top of the
+Overview page:
+
+![Instance ID](../../../img/database-access/guides/cloudsql/instance-id-mysql@2x.png)
+
+- Cloud SQL instance endpoint.
+
+You will use the instance's public IP address to connect to it. It can be viewed
+on the "Connect to this instance" panel on the Overview page:
+
+![Instance Public IP](../../../img/database-access/guides/cloudsql/instance-public-ip@2x.png)
+
+- Cloud SQL instance root certificate.
+
+The instance's root certificate is required so Teleport can validate the
+certificate presented by the database instance. You can download `server-ca.pem`
+file from the Connections tab under Security section:
+
+![Instance Root Certificate](../../../img/database-access/guides/cloudsql/instance-root-ca@2x.png)
+
+## Setup Teleport Auth and Proxy Services
+
+Teleport Database Access supports Cloud SQL MySQL starting from `7.0` release.
+Download the appropriate version of Teleport for your platform from our
+[downloads page](https://goteleport.com/teleport/download).
+
+Teleport requires a valid TLS certificate to operate and can fetch one automatically
+using Let's Encrypt [ACME](https://letsencrypt.org/how-it-works/) protocol.
+
+We will assume that you have configured DNS records for `teleport.example.com`
+and `*.teleport.example.com` to point to the Teleport node.
+
+Let's generate a Teleport config with ACME enabled:
+
+```bash
+$ teleport configure --cluster-name=teleport.example.com --acme --acme-email=alice@example.com -o file
+```
+
+<Admonition
+  type="note"
+  title="Web Proxy Port"
+>
+  Teleport uses [TLS-ALPN-01](https://letsencrypt.org/docs/challenge-types/#tls-alpn-01)
+  ACME challenge to validate certificate requests which only works on port `443`.
+  As such, in order to use ACME for certificate management, web proxy needs to
+  be accessible on port `443`.
+</Admonition>
```

Comment:
As per @inertial-frame it'd be good if this whole ACME section was something included rather than duplicated content, as I know we have these same instructions elsewhere in the docs and maintenance will be cumbersome when they all need to be updated individually.